core/meson.build

```diff
@@ -40,7 +40,7 @@ if cpp_compiler.get_id() == 'emscripten'
   wasm_exported_runtime_methods = '-sEXPORTED_RUNTIME_METHODS=[\'UTF8ToString\',\'stringToNewUTF8\']'
 endif
 
-subdir('doc')
+subdir('docs/internal')
```

Comment:
Is this causing  a CI failure?

> meson.build:43:0: ERROR: Nonexistent build file 'docs/internal/meson.build'

Reply:
@ermshiperete do you know what is causing this?

Reply:
`linux/scripts/dist.sh` ignores directories named `docs`. Intention was to ignore it on the root level, but I think we use the wrong syntax for that so that it gets ignored everywhere. Changing it to `--tar-ignore=./docs` would probably work.

Reply:
```
dpkg-source --tar-ignore=*~ \
    --tar-ignore=./.git \
    --tar-ignore=./.gitattributes \
    --tar-ignore=./.gitignore \
    --tar-ignore=experiments \
    --tar-ignore=debian \
    --tar-ignore=./.github \
    --tar-ignore=./.vscode \
    --tar-ignore=./android \
    --tar-ignore=./.devcontainer \
    --tar-ignore=./artifacts \
    \
    --tar-ignore=./common/models \
    --tar-ignore=./common/resources \
    --tar-ignore=./common/schemas \
    --tar-ignore=./common/test/keyboards/build.* \
    --tar-ignore=./common/test/resources \
    --tar-ignore=./common/web \
    --tar-ignore=./common/windows \
    \
    --tar-ignore=./core/build \
    --tar-ignore=./developer \
    --tar-ignore=./docs \
    --tar-ignore=./ios \
    --tar-ignore=./linux/keyman-config/keyman_config/version.py \
    --tar-ignore=./linux/keyman-config/buildtools/build-langtags.py \
    --tar-ignore=__pycache__ \
    --tar-ignore=./linux/docs/help \
    --tar-ignore=./mac \
    --tar-ignore=node_modules \
    --tar-ignore=./oem \
    --tar-ignore=./linux/build \
    --tar-ignore=./linux/builddebs \
    --tar-ignore=./linux/ibus-keyman/build \
    --tar-ignore=./linux/keyman-system-service/build \
    --tar-ignore=./resources/devbox \
    --tar-ignore=./resources/environment.sh \
    --tar-ignore=./resources/git-hooks \
    --tar-ignore=./resources/scopes \
    --tar-ignore=./resources/build/*.lua \
    --tar-ignore=./resources/build/jq* \
    --tar-ignore=./results \
    --tar-ignore=./tmp \
    --tar-ignore=./web \
    --tar-ignore=./windows \
    --tar-ignore=keyman_1* \
    --tar-ignore=dist \
    --tar-ignore=VERSION \
    \
    -Zgzip -b .
```

I am guessing a little here. What do you think?

Reply:
Looks good.

Reply:
>shame we can't do a whitelist in dist.sh?

Yes, unfortunately `dpkg-source` only supports `--tar-ignore` but doesn't have anything for whitelisting. I guess we could enhance our `dist.sh` script to build the tar-ignore parameters from a whitelist by adding `--tar-ignore` for all files and directories that are not in the whitelist, but that requires some work.

Reply:
Feels like, at this point, a whitelist of files to copy into a temp folder and dpkg-source from there would be cleaner.